### PR TITLE
[GEOT-7109] Swt Module review and update 

### PR DESCRIPTION
**Pull request by moovida**
[GEOT-7109](https://osgeo-org.atlassian.net/browse/GEOT-7109) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

This PR upgrades the SWT module as per discussions had in the mailinglist. 

This basically upgrades swt libraries, updates to latest gt api and fixes the bugs reported in JIRA.

After some discussion, it has been decided to remove the  RCP related part, since it is too complex to maintain (it would need an existing RCP project in a different repository). Also in the docs these parts were removed.
  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [x] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

### CODE REVIEW

**Comment by IanMayo** [comment]
Thanks @moovida , you've done a good job. I'm surprised at the volume of SWT changes - but I guess you've done some general tidying/refactoring too, right?

Is it an option to distribute the sample shapefile with the tutorial?  Loading/displaying the coastlines does give an "acknowledgement" that data has been loaded/rendered, IMHO.

**Comment by moovida** [comment]
Hi @IanMayo , yes there have been a couple of quick refactorings to cleanup and give the possibility to handle actions.

I am not sure how you mean to distribute a shp sample? You would like to be able to run the example and have a shp directly loaded? Or a sample to load with the "open shapefile" option?

**Comment by IanMayo** [comment]
Hello @moovida , for the tutorial element of the PR, it would seem useful to finish with some loaded data - since this demonstrates:
- data on the plot
- items in the layers panel
- a working EPSG

I see there is already a shapefile distributed in `docs\src\main\resources\org\geotools\grid`. Is it an option to include the `coastlines` shapefile that you originally used in the tutorial?   (I recognise the original tutorial was a _demonstration_ of you loading a shapefile from your local PC - but this would be more of a series of steps for a potential user to load/view the data).

image

**Comment by moovida** [comment]
Hi @IanMayo , I like the idea. We could also show how a "load default shp" action is added:

```
/**
 * Action to open a shapefile from a geotools source code location.
 */
public class LoadDefaultShpAction extends MapAction implements ISelectionChangedListener {

    public LoadDefaultShpAction() {
        super("Load default SHP", "Load the default geotools installation shapefile.", null);
    }

    public void run() {
        try {
            MapContent mapContent = mapPane.getMapContent();
            
            File shapeFile = new File("........geotools/docs/src/main/resources/org/geotools/grid/oz.shp");
            ShapefileDataStore store = new ShapefileDataStore(shapeFile.toURI().toURL());
            SimpleFeatureSource featureSource = store.getFeatureSource();
            SimpleFeatureCollection shapefile = featureSource.getFeatures();
            Style style = Utils.createStyle(shapeFile, featureSource);
            mapContent.addLayer(new FeatureLayer(shapefile, style));

            mapPane.redraw();
        } catch (Exception e) {
            e.printStackTrace();
        }

    }

    public void selectionChanged( SelectionChangedEvent arg0 ) {
    }
}
```

Which would be nice for the user to load something right away.

Or, we could just show how to load a shp at startup.

My main issue is that a user that is trying this is not forced to be doing this from the geotools source. In fact I imagine most people doing this from maven geotools. Therefore a shp from the repo might not be an option.

Should we just tell in the tutorial to place the path to an existing shapefile on the user's disk?

**Review by mprins — CHANGES_REQUESTED** [review]
please run the formatting tools on the code and the pom file; specifically the pom file was formatted according to the project rules, but that is destroyed with this PR.

You can use `mvn clean install -T1.1C -f modules/unsupported/swt/pom.xml `

**Comment by moovida** [comment]
> please run the formatting tools on the code and the pom file; specifically the pom file was formatted according to the project rules, but that is destroyed with this PR.
> 
> You can use `mvn clean install -T1.1C -f modules/unsupported/swt/pom.xml `

Thanks a lot for this.

File: modules/unsupported/swt/pom.xml [hunk 1]
```diff
@@ -1,139 +1,169 @@
...
+		</dependency>
+
+		<!-- https://mvnrepository.com/artifact/org.eclipse.platform/org.eclipse.equinox.common -->
+		<dependency>
+			<groupId>org.eclipse.platform</groupId>
+			<artifactId>org.eclipse.equinox.common</artifactId>
+			<version>3.15.0</version>
+		</dependency>
 
+		<!-- https://mvnrepository.com/artifact/org.eclipse.platform/org.eclipse.swt -->
```
[the remark this answers is not in the document]
Comment (moovida) [reply modules/unsupported/swt/pom.xml]: Ok, removed. Thanks.

**Comment by moovida** [comment]
> oops, of course. Just in case the "student" doesn't work with shapefiles (and may not have one handy), but still wants to complete the tutorial. I _guess_ we could provide a URL for a shapefile that we can expect to be persistent - such as: https://www.naturalearthdata.com/downloads/110m-physical-vectors/

@IanMayo, let me know what you think. I can make it even more tutorial-verbose, but I think this might be out of scope.

**Review by mprins — APPROVED** [review]

**Comment by IanMayo** [comment]
> @IanMayo, let me know what you think. I can make it even more tutorial-verbose, but I think this might be out of scope.

The new version, with OSM & Shapefiles is great - thanks.

**Review by IanMayo — APPROVED** [review]
Appreciate both the fixes/updates and the more comprehensive tutorial.

**Comment by aaime** [comment]
Added labels for automatic backport (I think you might want to backport this right?)

**Comment by aaime** [comment]
Well it run, but found conflicts... looks like a backport will have to be done manually, if you plan to do one.

**Comment by moovida** [comment]
Hi @aaime , thank you. 
It is strange, because I read about formatting issues. Could that have something to do with the fact that I initially scrambled the formatting and fixed it on the last commits? It should not, since it is squashed, right?

**Comment by aaime** [comment]
Correct, it should not.... try to do a cherry-pick and find out what is the source of the conflict :-D

**Comment by moovida** [comment]
Well, I tried the manual backport and fixed the small issues it had for 26.x

Am I supposed to do this?
```
git push --set-upstream origin backport-3921-to-26.x
```
even if along the way it took the original geotools repo as upstream? It is the first time I try to do a backport.

**Comment by moovida** [comment]
In my case origin is my repo, so the push should be there right (the set-upstream should take care of it in the push?)?
Sorry, I just do not want to break stuff :-)

Yes, 27 is next, if I am able to close 26.

**Comment by moovida** [comment]
Ok, the backport PR: https://github.com/geotools/geotools/pull/3926
shows me zillions of conflicting files. Not even sure how that can happen, but if this need much more care, then I will have to lay it down for now. 
Not even sure @IanMayo is interested in a backport, even if it would sure be handy in general.

**Comment by IanMayo** [comment]
Hello @moovida 

No, the backport won't bring value to me.  If it _just worked_, then that would be fine.  If it's troublesome then my needs are met with the `28.x` version.

**Comment by moovida** [comment]
Ok, I was just trying to go against the wrong branch. In fact it should work now. Were really small fixes. The 2 PR are now open for 26 and 27.

**Pull request merged**